basicswap/interface/firo.py

```diff
@@ -102,6 +102,75 @@ def isAddressMine(self, address: str, or_watch_only: bool = False) -> bool:
 
+    def getNewSparkAddress(self) -> str:
+        try:
+            return self.rpc_wallet("getnewsparkaddress")[0]
+        except Exception as e:
+            self._log.error(f"getnewsparkaddress failed: {str(e)}")
+            raise
+
+    def getNewStealthAddress(self):
+        """Get a new Spark address (alias for consistency with other coins)."""
+        return self.getNewSparkAddress()
+
+    def getWalletInfo(self):
+        """Get wallet info including Spark balance."""
+        rv = super(FIROInterface, self).getWalletInfo()
+        try:
+            spark_balance_info = self.rpc("getsparkbalance")
+            # getsparkbalance returns amounts in atomic units (satoshis)
+            # Field names: availableBalance, unconfirmedBalance, fullBalance
+            confirmed = spark_balance_info.get("availableBalance", 0)
+            unconfirmed = spark_balance_info.get("unconfirmedBalance", 0)
+            full_balance = spark_balance_info.get("fullBalance", 0)
+            # Values are already in atomic units, keep as integers
+            # basicswap.py will format them using format_amount
+            rv["spark_balance"] = confirmed if confirmed else 0
+            rv["spark_unconfirmed"] = unconfirmed if unconfirmed else 0
+            immature = full_balance - confirmed - unconfirmed
+            rv["spark_immature"] = immature if immature > 0 else 0
+        except Exception as e:
+            self._log.warning(f"getsparkbalance failed: {str(e)}")
+            rv["spark_balance"] = 0
+            rv["spark_unconfirmed"] = 0
+            rv["spark_immature"] = 0
+        return rv
+
+    def withdrawCoin(self, value, type_from: str, addr_to: str, subfee: bool) -> str:
+        """Withdraw coins, supporting both transparent and Spark transactions.
+
+        Args:
+            value: Amount to withdraw
+            type_from: "plain" for transparent, "spark" for Spark
+            addr_to: Destination address
+            subfee: Whether to subtract fee from amount
+        """
+        type_to = "spark" if addr_to.startswith("sm1") else "plain"
+
+        if "spark" in (type_from, type_to):
+            # RPC format: spendspark {"address": {"amount": ..., "subtractfee": ..., "memo": ...}}
+            # RPC wrapper will serialize this as: {"method": "spendspark", "params": [{...}], ...}
+            try:
+                # Construct params: dict where address is the key, wrapped in array for RPC
+                params = [{addr_to: {"amount": value, "subtractFee": subfee}}]
+
+                if type_from == "spark":
+                    result = self.rpc_wallet("spendspark", params)
+                else:
+                    result = self.rpc_wallet("mintspark", params)
+
+                # spendspark returns a txid string directly or in a result dict
+                if isinstance(result, dict):
+                    return result.get("txid", result.get("tx", ""))
+                return result
```

Comment:
```suggestion
                return result[0]
```
The txid is printing as `['txid_here']`

This works, i think? prints like `txid_here` (without the quotes and brackets)